solana/solana-ibc/programs/solana-ibc/src/tests.rs

```diff
@@ -643,10 +643,10 @@ fn anchor_test_deliver() -> Result<()> {
 
     let denom = format!("{}/{channel_id_on_a}/{send_denom}", port_id.clone());
     let hashed_denom = CryptoHash::digest(send_denom.as_bytes());
-    let hashed_base_denom =
+    let base_denom =
         ibc::apps::transfer::types::BaseDenom::from_str(&denom).unwrap();
```

Comment:
I’m looking into this right now and I’m pretty sure this should be
something along the lines of:

```rust
    let base_denom = mint_keypair.pubkey().to_string();
    let hashed_denom = CryptoHash::digest(base_denom.as_bytes());
    let denom = format!("{port_id}/{channel_id_on_a}/{base_denom}");
    let denom = ibc::apps::transfer::types::PrefixedDenom::from_str(&denom).unwrap();
    let token = ibc::apps::transfer::types::Coin {
        denom,
        amount: TRANSFER_AMOUNT.into(),
    };
```

(notice change from BaseDonem::from_str to PrefixedDenom::from_str).
However, this does break the test.

Reply:
it makes sense to use `PrefixedDenom` instead of `BaseDenom`. It is good that the test is failing because since we were using `BaseDenom`, it was assuming that the packet is sent on source chain and unescrowing tokens but as per the current denom, it should have burned tokens instead. I have made the changes and it seems to be working.